src/cml.js

```diff
@@ -477,7 +478,7 @@ class CML {
         await exec('git fetch --unshallow');
       }
     }
-    await exec('git fetch --all');
+    await exec(`git fetch --all --depth=${fetchDepth}`);
```

Comment:
git fetch doesn't like 0 so lets check for this condition.

```suggestion
    if (fetchDepth === 0) {
      return await exec('git fetch --all --unshallow');
    }
    return await exec(`git fetch --all --depth=${fetchDepth}`);
```